packages/hydrojudge/src/sandbox.ts

```diff
@@ -1,50 +1,79 @@
-import Axios from 'axios';
 import cac from 'cac';
 import fs from 'fs-extra';
+import { ParseEntry } from 'shell-quote';
 import { STATUS } from '@hydrooj/utils/lib/status';
 import { getConfig } from './config';
 import { FormatError, SystemError } from './error';
 import { Logger } from './log';
+import { SandboxClient } from './sandbox_client';
+import {
+    Cmd, CopyInFile, SandboxResult, SandboxStatus,
+} from './sandbox_interface';
 import { cmd, parseMemoryMB } from './utils';
 
 const argv = cac().parse();
 const logger = new Logger('sandbox');
 let callId = 0;
 let supportOptional = false;
 
-const statusMap = {
-    'Time Limit Exceeded': STATUS.STATUS_TIME_LIMIT_EXCEEDED,
-    'Memory Limit Exceeded': STATUS.STATUS_MEMORY_LIMIT_EXCEEDED,
-    'Output Limit Exceeded': STATUS.STATUS_RUNTIME_ERROR,
-    Accepted: STATUS.STATUS_ACCEPTED,
-    'Nonzero Exit Status': STATUS.STATUS_RUNTIME_ERROR,
-    'Internal Error': STATUS.STATUS_SYSTEM_ERROR,
-    'File Error': STATUS.STATUS_SYSTEM_ERROR,
-    Signalled: STATUS.STATUS_RUNTIME_ERROR,
-};
+const statusMap: Map<SandboxStatus, number> = new Map([
+    [SandboxStatus.TimeLimitExceeded, STATUS.STATUS_TIME_LIMIT_EXCEEDED],
+    [SandboxStatus.MemoryLimitExceeded, STATUS.STATUS_MEMORY_LIMIT_EXCEEDED],
+    [SandboxStatus.OutputLimitExceeded, STATUS.STATUS_RUNTIME_ERROR],
+    [SandboxStatus.Accepted, STATUS.STATUS_ACCEPTED],
+    [SandboxStatus.NonzeroExitStatus, STATUS.STATUS_RUNTIME_ERROR],
+    [SandboxStatus.InternalError, STATUS.STATUS_SYSTEM_ERROR],
+    [SandboxStatus.FileError, STATUS.STATUS_SYSTEM_ERROR],
+    [SandboxStatus.Signalled, STATUS.STATUS_RUNTIME_ERROR],
+]);
+
+interface Parameter {
+    time?: number;
+    stdin?: string;
+    stdout?: string;
+    stderr?: string;
+    execute?: string;
+    memory?: number;
+    processLimit?: number;
+    copyIn?: { [dst: string]: CopyInFile };
+    copyOut?: string[];
+    copyOutCached?: string[];
+}
+
+function checkStringArray(args: ParseEntry[]): args is string[] {
+    return args.every((arg: ParseEntry) => typeof arg === 'string');
+}
+
+function parseArgs(execute: string): string[] {
+    const args = cmd(execute.replace(/\$\{dir\}/g, '/w'));
+    if (!checkStringArray(args)) {
+        throw new FormatError(`${execute} contains invalid operator`);
```

Comment:
FormatError means wrong testdata or configuration format, this is system language definition issue and SystemError should be thrown.